client/my-sites/themes/index.node.js

```diff
@@ -15,9 +15,13 @@ export default function( router ) {
 
 	if ( config.isEnabled( 'manage/themes' ) ) {
 		if ( config.isEnabled( 'manage/themes-ssr' ) ) {
-			router( `/design/:vertical(${ verticals })?/:tier(free|premium)?$`, fetchThemeDataWithCaching, loggedOut, makeLayout );
-			router( `/design/:vertical(${ verticals })?/:tier(free|premium)?`, fetchThemeData, loggedOut, makeLayout );
-			router( `/design/:vertical(${ verticals })?/:tier(free|premium)?/filter/:filter`, fetchThemeData, loggedOut, makeLayout );
+			router( `/design/:vertical(${ verticals })?/:tier(free|premium)?`, fetchThemeDataWithCaching, loggedOut, makeLayout );
```

Comment:
> This will now cache arbitrary searches.

Arbitrary combinations of tiers/filters/verticals, but [not querystrings](https://github.com/Automattic/wp-calypso/pull/12592/files#diff-cdf8be48e336c794267be91d81a3a610R141). (The `false` arg [here](https://github.com/Automattic/wp-calypso/pull/12592/files#diff-cdf8be48e336c794267be91d81a3a610R143) is [`shouldUseCache`](https://github.com/Automattic/wp-calypso/pull/12592/files#diff-cdf8be48e336c794267be91d81a3a610R115)).

> Seems we might want to increase the 500 slot LRU?

500 might be enough for those combinations -- before this change, I think we were really only caching the no-search case, so we were probably only using one slot.